pkg/operator/ceph/object/cosi/spec.go

```diff
@@ -97,11 +97,13 @@ func createCOSIPodSpec(cephCOSIDriver *cephv1.CephCOSIDriver) (corev1.PodTemplat
 
 	podTemplateSpec := corev1.PodTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:   CephCOSIDriverName,
-			Labels: getCOSILabels(cephCOSIDriver.Name, cephCOSIDriver.Namespace),
+			Name:        CephCOSIDriverName,
+			Labels:      getCOSILabels(cephCOSIDriver.Name, cephCOSIDriver.Namespace),
+			HostNetwork: CephCOSIDriver.Spec.Network.IsHostStrict(),
```

Comment:
CephCOSIDriver's Spec doesn't have a `Network` spec or field. I think only CephCluster has a `Network` spec. CephObjectStore and CephNFS have a `hostNetwork` boolean to allow controlling those on a per-resource basis if desired; otherwise, those default to the host network configuration of the CephCluster.

When dealing with `host-strict`, non-CephCluster Rook controllers will have to make sure to get the related CephCluster (most of them do already at the highest level) and check its `Network` config. This would be things like the RBD/CephFS mirror controllers.

---

But I'm realizing that there is a bigger issue here that I didn't think of before: the CSI and COSI drivers. Both CSI and COSI are k8s-cluster-wide resources and not per-CephCluster resources.

I think we will need to add more places where the host network behavior is controlled for those k8s-cluster-scope controllers. 

There is a `CSI_ENABLE_HOST_NETWORK=<true|false>` config that already exists. I think we could go one of two directions here:
1. Add a new config var like `CSI_FORCE_HOST_NETWORK = <true|false>`
2. Add a third option to the existing var `CSI_ENABLE_HOST_NETWORK = 'force'`

Personally, I think option 2 is preferable so that users dont' get confused if `_FORCE_` is true but `ENABLE` is false. But it might make the helm charts more complicated. Let's solicit @Madhu-1 's input here.

For COSI, I think it makes sense to add a new `hostNetwork: <true|false>` field to its CRD, and allow users to set that option. 

I am also wondering here about the discover daemonset. Is that run per-CephCluster or per-Operator? @travisn do you remember? If it's per-Operator (like CSI), we may also need to follow the CSI plan from above here also.

Reply:
I spoke with Travis. We think it makes most sense to pivot away from creating a new `"host-strict"` and instead use a `rook-ceph-operator-config` configmap value like `ROOK_FORCE_HOST_NETWORK`. 

The way that it's planned now, we would have to implement the new `host-strict` mode **AND** an operator config in addition, which is a fair bit of complexity. We think it makes the most sense to only use an operator config. For the use case we are trying to enable, this makes the most sense anyway: if a k8s cluster has no CNI, an operator-level (global as far as Rook is concerned) config is the best single location to set force use of host networking.

Apologies for the false start in adding a new network.provider. I still think that using `IsHost(/* plus configmap config input */)` is the best way to ensure that existing usage of `IsHost()` continues to operate as expected when the newly-planned operator config is set.